tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py

```diff
@@ -2708,6 +2708,68 @@ def visit_importfrom(self, node):
                     )
 
 
+class DoNotLogErrorsEndUpRaising(BaseChecker):
+
+    """Rule to check that errors that get raised aren't logged"""
+
+    name = "do-not-log-raised-errors"
+    priority = -1
+    msgs = {"C4762": (
+            "Do not log errors that get raised in an exception block.",
+            "do-not-log-raised-errors",
+            "Do not log errors at error or warning level when error is raised in an exception block",
+            ),
+            }
+
+    def visit_try(self, node):
+        """Check that raised errors aren't logged at 'error' or 'warning' levels in exception blocks.
+           Go through exception block and branches and ensure error hasn't been logged if exception is raised.
+        """
+        # Return a list of exception blocks
+        except_block = node.handlers
+        # Iterate through each exception block
+        for nod in except_block:
+            # Get the nodes in each block (e.g. nodes Expr and Raise)
+            exception_block_body = nod.body
+            self.check_for_raise(exception_block_body)
+
+    def check_for_raise(self, node):
+        """ Helper function - checks for instance of 'Raise' node
+            Also checks 'If' and nested 'If' branches
+        """
+        for i in node:
+            if isinstance(i, astroid.Raise):
+                self.check_for_logging(node)
+            # Check for any nested 'If' branches
+            if isinstance(i, astroid.If):
+                self.check_for_raise(i.body)
+                # Check any 'elif' and 'else' branches
+                elif_statements = i.orelse
+                while len(elif_statements) == 1:
+                    if isinstance(elif_statements[0], astroid.If):
+                        for x in elif_statements:
+                            self.check_for_raise(x.body)
+                            elif_statements = x.orelse
```

Comment:
I like the recursive approach here - though it's generally best to avoid reassigning the iterable while it's currently being iterated. Looking at the tests this seems to be functioning correctly - if you could link me to the docs for the `astroid.If` node and what the structure of the `orelse` property is I can see if I have any ideas.

Reply:
Hi, the `If` node docs can be found [here](https://pylint.readthedocs.io/projects/astroid/en/latest/api/nodes/astroid.nodes.If.html), `orelse` can be found on this page too. The `orelse` property is a list of nodes as described in the above comment :)

Reply:
Thank you! I did some playing around with this and I figure out we don't actually need to loop over the contents of the `orelse` property - we can actually pass the entire thing into back into the `check_for_raise` function and it will run the iteration for us :)

So I think the whole thing can be simplified into this:
```python
    def check_for_raise(self, node):
        """ Helper function - checks for instance of 'Raise' node
            Also checks 'If' and nested 'If' branches
        """
        for i in node:
            if isinstance(i, astroid.Raise):
                self.check_for_logging(node)
                
            # Check for any nested 'If' branches
            if isinstance(i, astroid.If):
                self.check_for_raise(i.body)

                # Check any 'elif', 'else' or 'for' branches
                self.check_for_raise(i.orelse)
```
Let me know if this has any issues with the testing!

Reply:
Ah yes! Works perfectly thank you! Think I overcomplicated it in my head :)